eve/methods/common.py

```diff
@@ -850,27 +850,25 @@ def sort_db_response(embedded_docs, id_value_to_sort, list_of_id_field_name):
     return temp_embedded_docs
 
 
-def sort_per_resource(embedded_docs, id_value_to_sort, id_field_name):
+def sort_per_resource(embedded_docs, id_values_to_sort, id_field_name):
     """ Sorts the documents fetched from the database per single resource
 
-        :param embedded_docs: the documents fetch from the database.
-        :param id_value_to_sort: id_value sort criteria.
+        :param embedded_docs: list of the documents fetched from the database.
+        :param id_values_to_sort: list of the id_values sort criteria.
         :param list_of_id_field_name: list of name of fields
+        :param id_field_name: key name of the id field; `_id`
         :return embedded_docs: the list of documents sorted as per input
         """
-    # Removing None
-    number_of_none = embedded_docs.count(None)
-    if number_of_none:
-        embedded_docs = [x for x in embedded_docs if x is not None]
+    if isinstance(id_values_to_sort, list) and id_values_to_sort is None:
```

Comment:
won't this always be false? you probably only mean `if id_values_to_sort is None:`